src/uimodel.cpp

```diff
@@ -2210,6 +2216,21 @@ std::string UiModel::GetLastMessageId(const std::string& p_ProfileId, const std:
   return lastMessageId;
 }
 
+ChatInfo* UiModel::GetChatInfo(const std::string& p_ProfileId, const std::string& p_ContactId)
```

Comment:
@d99kris 

I couldn't find another public function returning a pointer in the codebase.
Maybe this is not a good practice? (I'm new to C++)

Anyway, this probably should be changed to a smart pointer.

Reply:
Good observation - yeah, generally speaking it's considered good practise to avoid raw pointers in modern C++ code. Smart pointers can often be used instead, however in this particular case we might need to return ChatInfo by value. 

The reason being that we're accessing a data structure `m_ChatInfos` shared across threads, and once we leave the function scope (and release the mutex), some other thread may modify/remove the entry we have a pointer to.

But before making any changes to the PR, please let me take a look at the full patch and share my feedback.